druid/src/widget/svg.rs

```diff
@@ -0,0 +1,284 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An SVG widget.
+
+use crate::{
+    BaseState, BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, PaintCtx, Size, UpdateCtx,
+    Widget,
+};
+
+use log::error;
+use std::marker::PhantomData;
+
+#[cfg(feature = "svg")]
+use crate::{
+    kurbo::BezPath,
+    piet::{Color, RenderContext},
+    Point,
+};
+
+#[cfg(feature = "svg")]
+pub use usvg;
```

Comment:
Reexporting is to cover the use case of someone wanting do something custom with SVGs while relying on my svg-to-piet function (which expects a usvg tree). If people have to import usvg in their own cargo.toml to do stuff that could cause conflicts yes?

Reply:
it would only cause conflicts if they were using a specific version of `usvg` that was different from the one that we support. I think it's okay to not re-export; presumably somebody using this functionality can control the version of `usvg` that they support, and if there are future major releases it shouldn't be hard for us to update.

Reply:
I think reexporting crates that appear in the public API is reasonably good practice to reduce breakage. Better still might be hiding the external dep entirely with a newtype, though.

Reply:
Okay I did the obvious-in-hindsight thing and just exported the functions we need to make a tree out of a str or a path.